container/service.proto

```diff
@@ -99,3 +105,44 @@ message ListResponse {
     // CID (container id) is list of SHA256 hashes of the container structures
     repeated bytes CID = 1 [(gogoproto.customtype) = "CID", (gogoproto.nullable) = false];
 }
+
+message ExtendedACLKey {
+    // ID (container id) is a SHA256 hash of the container structure
+    bytes ID = 1 [(gogoproto.customtype) = "CID", (gogoproto.nullable) = false];
+}
+
+message ExtendedACLValue {
+    // EACL is a binary representation of extended ACL rule table
+    bytes EACL = 1;
+
+    // Signature carries EACL signature
+    bytes Signature = 2;
+}
+
+message SetExtendedACLRequest {
+    ExtendedACLKey Key = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+
+    // Value carries extended ACL information
+    ExtendedACLValue Value = 2 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+
+    // RequestMetaHeader contains information about request meta headers (should be embedded into message)
+    service.RequestMetaHeader Meta           = 98 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+    // RequestVerificationHeader is a set of signatures of every NeoFS Node that processed request (should be embedded into message)
+    service.RequestVerificationHeader Verify = 99 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+}
+
+message SetExtendedACLResponse {}
+
+message GetExtendedACLRequest {
+    ExtendedACLKey Key = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+
+    // RequestMetaHeader contains information about request meta headers (should be embedded into message)
+    service.RequestMetaHeader Meta           = 98 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+    // RequestVerificationHeader is a set of signatures of every NeoFS Node that processed request (should be embedded into message)
+    service.RequestVerificationHeader Verify = 99 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
+}
+
+message GetExtendedACLResponse {
+    // ACL carries extended ACL information
+    ExtendedACLValue ACL = 2 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
```

Comment:
`ExtendedACLValue` contains signature of extended ACL, but I don't think smart-contract will store it.
Signature is required to check permission of `SetEACL` method in smart-contract. As soon as permission checked, 
signature is dropped.

Reply:
@AlexVanin, I thought it would be advisable to attach the signature of the owner of the container as proof. In any case, it can be left blank.

Reply:
@cthulhu-rider Yes, but if blockchain is a transparent source of truth for everything in the storage network, there is no need in additional verifications on `get`. If signature field will be empty there, it is worth to mention it at inline docs.

Reply:
After brief discussion with @anatoly-bogatyrev @realloc @im-kulikov we decided to keep signatures in the smart-contract to avoid attacks on neofs-node & morph communication.

Reply:
@AlexVanin, I agree, blockchain is a source of truth, but rpc is not. Actually, call and messages does not imply an appeal to the blockchain. Even if it does, there is a potential privilege spoofing vulnerabilities through rpc.